AGENTS.md

```diff
@@ -0,0 +1,38 @@
+# BoardwalkCash Agent Guidelines
+
+## Project Overview
+
+- **Framework**: Remix on Express server
+- **Development**: Nix + devenv, Bun for scripts/tests
+- **Database**: Supabase with migrations in `supabase/migrations`
+
+## Directory Structure
+
+- `app/components/` – Reusable UI components
+- `app/lib/` – Utilities and server helpers
+- `app/features/` – Feature-specific code (see `GUIDELINES.md`)
+- `app/routes/` – Remix routes
+- `e2e/` – Playwright tests
+- `supabase/`
+    ├── migrations/           # Database migration files
+    ├── database.types.ts     # Generated TypeScript types
+    └── config.toml          # Supabase configuration
+
+Use **kebab-case** for files/directories. Follow hierarchy in `GUIDELINES.md`.
+
+## Development
+
+- Run `bun run fix:all` for lint, format, and type checks
+- After schema changes: `bun run db:generate-types`
+- Tests: `bun test` (unit), `bun run test:e2e` (e2e)
+
+## React Guidelines
+
+- Use hooks for side effects and data fetching
+- Use **@tanstack/react-query** for network requests (`useQuery`, `useSuspenseQuery`, `useMutation`)
+- Avoid `useEffect` for data fetching
```

Comment:
these three items are a bit confusing. it says use hooks for side effect and data fetching and then it says rect query for network and then don't use useEffect for fetching

useEffect is the only one that should be used for side effects. all other hooks can/should not have side effects. data fetching/network request is a side effect but whenever possible we should do that with tanstack query instead of from useEffect

additionally I would mnetion to keep components mostly about ui and move logic to hooks